src/packet-ffxiv.c

```diff
@@ -103,9 +107,9 @@ static int dissect_message(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree,
     TODO: insert code for dealing with message types,
     for now register it as generic data.
   */
-  add_new_data_source(pinfo, tvb, "Message Data");
+  add_new_data_source(pinfo, tvb_new_subset_length(tvb, 0, header.length), "Message Data");
 
-  return tvb_captured_length(tvb);
+  return header.length;
```

Comment:
no particular reason, but since this subset is created solely for adding new data source I don't see the reason to keep a pointer to it and returning header.length because that's how much we have processed. (assuming partial/malformed packets are dealt with already)